src/RefractDataStructure.cc

```diff
@@ -1081,6 +1146,20 @@ namespace drafter {
                 throw snowcrash::Error("unknown type of data structure", snowcrash::MSONError);
         }
 
+        CircularReferenceChecker cr(GetNamedTypesRegistry());
```

Comment:
I think it would be more readable if this was named `checker` or `referencedChecker` instead of `cr`.

```cpp
if (cr) {
```

vs

```cpp
if (checker) {
```